pkg/controller/certificates/controller.go

```diff
@@ -181,6 +187,51 @@ func (cc *CertificateController) maybeSignCertificate(key string) error {
 	}
 	csr := obj.(*certificates.CertificateSigningRequest)
 
+	func() {
+		if !cc.approveAllKubeletCSRs ||
```

Comment:
> What error are you seeing?

Well, the initial symptoms were that my client never gets the signed cert. I've added a log message inside that condition and more log message in this func, and all I have ever seen is that log message.

If you are asking about the leakage issue that my fix caused, this what I'm seeing:
```
> docker exec kubelet curl -s localhost:8080/apis/certificates/v1alpha1/certificatesigningrequests
{
  "kind": "CertificateSigningRequestList",
  "apiVersion": "certificates/v1alpha1",
  "metadata": {
    "selfLink": "/apis/certificates/v1alpha1/certificatesigningrequests",
    "resourceVersion": "6530"
  },
  "items": [
    {
      "metadata": {
        "name": "csr-iwdqm",
        "generateName": "csr-",
        "selfLink": "/apis/certificates/v1alpha1/certificatesigningrequests/csr-iwdqm",
        "uid": "4528274c-687e-11e6-815e-aea4fb56590d",
        "resourceVersion": "6521",
        "creationTimestamp": "2016-08-22T15:37:01Z"
      },
      "spec": {
        "request": "LS0tLS1CRUdJTiBDRVJUSUZJQ0FURSBSRVFVRVNULS0tLS0KTUlIdU1JR1VBZ0VBTURJeEZUQVRCZ05WQkFvVERITjVjM1JsYlRwdWIyUmxjekVaTUJjR0ExVUVBeE1RYzNsegpkR1Z0T201dlpHVTZWRTlFVHpCWk1CTUdCeXFHU000OUFnRUdDQ3FHU000OUF3RUhBMElBQkREV1hYcnFWc2VUCkR6S1pHejZDZ0lMWU9ueDA5ck5BZXRSVWpMNVVObGJoV3liN1ROR295Snp4alVIZVZHREVvN21sbyswaVhwNEUKcU1TWEVxNnNiOHVnQURBS0JnZ3Foa2pPUFFRREFnTkpBREJHQWlFQXlXSzJMUGNVNzZZeE5kcU54KzI1c2VwYgpCNFMya2JtSVhWNEdxb3VkNUxVQ0lRRHk1SmNkTGRnSzQycXRJY1BFZkR1bjJJbnBtMldDRk5uaEQ5dmZKb1R3CnhBPT0KLS0tLS1FTkQgQ0VSVElGSUNBVEUgUkVRVUVTVC0tLS0tCg==",
        "username": "kubeadm-node-csr",
        "uid": "0983d63a-687e-11e6-8475-0242ac110002",
        "groups": [
          "system:kubelet-bootstrap"
        ]
      },
      "status": {
        "conditions": [
          {
            "type": "Approved",
            "reason": "AutoApproved",
            "message": "Auto approving of all kubelet CSRs is enabled on the controller manager",
            "lastUpdateTime": null
          },
          {
            "type": "Approved",
            "reason": "AutoApproved",
            "message": "Auto approving of all kubelet CSRs is enabled on the controller manager",
            "lastUpdateTime": null
          },
          ... // it keep repeating the message above
```